src/Integration/Binding/CSharpVBBindingOperation.cs

```diff
@@ -127,19 +127,12 @@ private void AddRuleset()
 
         private void AddAdditionalFile()
         {
-            try
-            {
-                var projectSystem = serviceProvider.GetService<IProjectSystemHelper>();
-                projectSystem.AssertLocalServiceIsNotNull();
-                // It's possible that the additional file is set property, but the ruleset is out of date, which will trigger the binding. So we need to remove the file first, otherwise we get an error that the file already exists.
-                // It's possible that the additional file exists but has a wrong type, e.g. Compile/Content, and not AdditionalFiles. In which case removing and re-adding it solves the problem.
-                projectSystem.RemoveFileFromProject(initializedProject, cSharpVBBindingConfig.AdditionalFile.Path);
-                projectSystem.AddFileToProject(initializedProject, cSharpVBBindingConfig.AdditionalFile.Path, "AdditionalFiles");
-            }
-            catch (Exception ex)
-            {
-                throw new Exception($"Failed to bind project {initializedProject.FullName}, exception details: {ex}");
-            }
+            var projectSystem = serviceProvider.GetService<IProjectSystemHelper>();
+            projectSystem.AssertLocalServiceIsNotNull();
+            // It's possible that the additional file is set property, but the ruleset is out of date, which will trigger the binding. So we need to remove the file first, otherwise we get an error that the file already exists.
+            // It's possible that the additional file exists but has a wrong type, e.g. Compile/Content, and not AdditionalFiles. In which case removing and re-adding it solves the problem.
+            projectSystem.RemoveFileFromProject(initializedProject, Path.GetFileName(cSharpVBBindingConfig.AdditionalFile.Path));
```

Comment:
I encountered a few problems during testing:
1) the "remove" is removing based just on the file name, so it will remove any file called "SonarLint.xml". This seems odd, since we have a check elsewhere that is supposed to throw an exception and stop the binding if there is another "SonarLint.xml" file (although that check doesn't seem to be working);
2) attempting to remove file or change the item group value throws an exception if the reference to the file is in a `Directory.Build.props` file;
3) removing and re-adding the item can cause the entry to change location in the project file, which doesn't matter functionally but is irritating when diff-ing the file for changes.

It's simple enough to do a check to see if the correct file is already referenced using the full path since we already have a method for that, so we don't need to remove the file - if it doesn't exist we can add it, and if it does exist we can try to change the item group if necessary.

However, trying to change the item group will still fail if the entry is in an imported file, and I don't think we can easily do anything about that except make sure we catch the exception, fail the binding and warn the user.